Migrate snowflake crawler to using ACCOUNT [sc-10159] (#404)
MetaphorData/connectors: metaphor/snowflake/README.md

create role identifier($role) comment = 'Limited access role for Metaphor connector';
grant usage on warehouse identifier($warehouse) to role identifier($role);
grant usage on database identifier($db) to role identifier($role);
-- Grant permissions to access the snowflake "Account Usage" views:
grant imported privileges on database snowflake to role identifier($role);
-- If there are inbound shared databases, grant permissions to "show shares"

> grant import share on account to identifier($role);

Author: Yeah, it was a typo before, corrected here.

alter user identifier($user) set rsa_public_key='<public_key_content>';
```

### Fetching extra info (Optional)

> If you intend to fetch extra table information such as LAST_CHANGE_COMMIT_TIME or DDL (enabled by default), following additional grant needs to be given to the newly created role.

Reviewer: ```suggestion
To fetch extra table information such as the last modified time or DDL, please grant the following additional privileges to the newly created role.
```

```sql

> grant usage on all schemas in database identifier($db) to role identifier($role);

Reviewer: Were you able to verify the permission in the end?

Author: Verified the permission grants, both for basic info (don't need table grant) or extra info (require table grants).

See [Filter Configurations](../common/docs/filter.md) for more information on the optional `filter` config.

#### Fetch Extra Table Information

> By default, the crawler fetches extra table information such as LAST_CHANGE_COMMIT_TIME or DDL. This requires extra grant to all the tables. To disable this, add the following:

Reviewer: ```suggestion
By default, the connector fetches extra table information such as LAST_CHANGE_COMMIT_TIME or DDL. This requires additional grants to all the tables. The connector will print warning messages if it has insufficient permissions. To disable this feature altogether, add the following to your config:
```
